windows/devices/uwb/UwbDeviceConnector.cxx

```diff
@@ -194,7 +224,36 @@ UwbDeviceConnector::SessionDeinitialize(uint32_t sessionId)
 {
     std::promise<UwbStatus> resultPromise;
     auto resultFuture = resultPromise.get_future();
-    // TODO: invoke IOCTL_UWB_SESSION_DEINIT
+    
+    wil::shared_hfile handleDriver;
+    auto hr = OpenDriverHandle(handleDriver, m_deviceName.c_str());
+    if (FAILED(hr)) {
+        PLOG_ERROR << "failed to obtain driver handle for " << m_deviceName << ", hr=" << hr;
+        return resultFuture;
+    }
+
+    const UWB_SESSION_DEINIT sessionDeinit{
+        .size = sizeof(UWB_SESSION_DEINIT),
+        .sessionId = sessionId,
+    };
+    UWB_STATUS status;
+
+    // Determine the amount of memory required for the UWB_SESSION_DEINIT from the driver.
+    BOOL ioResult = DeviceIoControl(handleDriver.get(), IOCTL_UWB_SESSION_DEINIT, const_cast<UWB_SESSION_DEINIT*>(&sessionDeinit), sizeof sessionDeinit, &status, sizeof status, nullptr, nullptr);
+    if (!LOG_IF_WIN32_BOOL_FALSE(ioResult)) {
+        // TODO: need to do something different here
+        HRESULT hr = HRESULT_FROM_WIN32(GetLastError());
+        PLOG_ERROR << "error when sending IOCTL_UWB_SESSION_DEINIT, hr=" << std::showbase << std::hex << hr;
+        return resultFuture;

```

Comment:
Need to set an exception in resultPromise before returning (see earlier comment for details).